collector/info_schema_clientstats.go

```diff
@@ -0,0 +1,180 @@
+				"The number of ROLLBACK commands issued by this client’s connections.",
+				[]string{"client"}, nil)},
+		"DENIED_CONNECTIONS": {prometheus.CounterValue,
+			prometheus.NewDesc(prometheus.BuildFQName(namespace, informationSchema, "client_statistics_denied_connections_total"),
+				"The number of connections denied to this client.",
+				[]string{"client"}, nil)},
+		"LOST_CONNECTIONS": {prometheus.CounterValue,
+			prometheus.NewDesc(prometheus.BuildFQName(namespace, informationSchema, "client_statistics_lost_connections_total"),
+				"The number of this client’s connections that were terminated uncleanly.",
+				[]string{"client"}, nil)},
+		"ACCESS_DENIED": {prometheus.CounterValue,
+			prometheus.NewDesc(prometheus.BuildFQName(namespace, informationSchema, "client_statistics_access_denied_total"),
+				"The number of times this client’s connections issued commands that were denied.",
+				[]string{"client"}, nil)},
+		"EMPTY_QUERIES": {prometheus.CounterValue,
+			prometheus.NewDesc(prometheus.BuildFQName(namespace, informationSchema, "client_statistics_empty_queries_total"),
+				"The number of times this client’s connections sent empty queries to the server.",
+				[]string{"client"}, nil)},
+		"TOTAL_SSL_CONNECTIONS": {prometheus.CounterValue,
+			prometheus.NewDesc(prometheus.BuildFQName(namespace, informationSchema, "client_statistics_total_ssl_connections_total"),
+				"The number of times this client’s connections connected using SSL to the server.",
+				[]string{"client"}, nil)},
+	}
+)
+
+// ScrapeClientStat collects from `information_schema.client_statistics`.
+func ScrapeClientStat(db *sql.DB, ch chan<- prometheus.Metric) error {
+	informationSchemaClientStatisticsRows, err := db.Query(clientStatQuery)
+	if err != nil {
+		return err
+	}
+	defer informationSchemaClientStatisticsRows.Close()
+
+	// The client column is assumed to be column[0], while all other data is assumed to be coerceable to float64.
+	// Because of the client column, clientStatData[0] maps to columnNames[1] when reading off the metrics
+	// (because clientStatScanArgs is mapped as [ &client, &clientData[0], &clientData[1] ... &clientdata[n] ]
+	// To map metrics to names therefore we always range over columnNames[1:]
+	var columnNames []string
+	columnNames, err = informationSchemaClientStatisticsRows.Columns()
+	if err != nil {
+		return err
+	}
+
+	var client string                                        // Holds the client name, which should be in column 0.
+	var clientStatData = make([]float64, len(columnNames)-1) // 1 less because of the client column.
+	var clientStatScanArgs = make([]interface{}, len(columnNames))
+	clientStatScanArgs[0] = &client
+	for i := range clientStatData {
+		clientStatScanArgs[i+1] = &clientStatData[i]
+	}
+
+	for informationSchemaClientStatisticsRows.Next() {
+		err = informationSchemaClientStatisticsRows.Scan(clientStatScanArgs...)
+		if err != nil {
```

Comment:
Write in one row:

`if err := informationSchemaClientStatisticsRows.Scan(clientStatScanArgs...); err != nil {`